src/routers/alpha-router/alpha-router.ts

```diff
@@ -630,12 +630,26 @@ export class AlphaRouter
       }
       exchangeRate = swap.trade!.outputAmount.divide(swap.trade!.inputAmount);
 
-      log.info({
-        optimalRatio: optimalRatio.asFraction.toFixed(18),
-        newRatio: newRatio.asFraction.toFixed(18),
-        ratioErrorTolerance: swapAndAddConfig.ratioErrorTolerance.toFixed(18),
-        iterationN: n.toString(),
-      });
+      log.info(
+        {
+          exchangeRate: exchangeRate.asFraction.toFixed(18),
+          optimalRatio: optimalRatio.asFraction.toFixed(18),
+          newRatio: newRatio.asFraction.toFixed(18),
+          inputBalanceUpdated: inputBalanceUpdated.asFraction.toFixed(18),
+          outputBalanceUpdated: outputBalanceUpdated.asFraction.toFixed(18),
+          ratioErrorTolerance: swapAndAddConfig.ratioErrorTolerance.toFixed(18),
+          iterationN: n.toString(),
+        },
+        'QuoteToRatio Iteration Parameters'
+      );
+
+      if (exchangeRate.equalTo(0)) {
```

Comment:
another option other than erroring is to trade what we do have, then add that as liquidity and keep the rest as dust. Alternatively, if the UI gets this error it could suggest trying to swap a smaller amount.